doc/nrf/releases/release-notes-latest.rst

```diff
@@ -64,12 +67,12 @@ The following list summarizes the most important changes inherited from upstream
 
   * MQTT:
 
-    * The ``utf8`` pointer in the :c:type:`mqtt_utf8` struct is now const.
+    * The ``utf8`` pointer in the :cpp:class:`mqtt_utf8` struct is now const.
```

Comment:
This is not a C++ class, it's a C struct. If this needs to be changed, `:c:struct:` would be more appropriate. Is this being copy/pasted from elsewhere in the documentation? If so, further cleanup may be worth doing.

https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#directive-c-struct

Reply:
This is a limitation in Breathe. When using :c:type:, the link will not work. That's why we need to use :cpp:class: instead - @utzig , can you elaborate?

Reply:
> This is not a C++ class, it's a C struct. If this needs to be changed, `:c:struct:` would be more appropriate. Is this being copy/pasted from elsewhere in the documentation? If so, further cleanup may be worth doing.
> 
> https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#directive-c-struct

This was introduced with Sphinx>3 and we are still using `2.4.4`. Also breathe parsed most Doxygen XML as if it was cpp until very recently. Using `:cpp:class`  is still the correct way to do the linking (as you must be quite used to seeing `:cpp:func`) until we upgrade to latest tooling.

Reply:
> as you must be quite used to seeing `:cpp:func`

I'm not used to this. For example `:c:func:` is used to link to devicetree macros throughout zephyr/doc/guides/dts/api-usage.rst. These work.

Grepping `zephyr/doc` for `:c:func:` has 668 matches total, many of which are to legitimate functions, not just macros. However, it looks like you're right that when linking to functions, these don't generate links.

This seems like a bug which should be fixed in the upstream zephyr docs, I guess. Or perhaps we can upgrade our tooling so the C domain works; using C++ sounds like it's a workaround.

Anyway, thanks for explaining, seems fine for now!